src/defineFunction.js

```diff
@@ -104,13 +106,14 @@ type FunctionDefSpec<NODETYPE: NodeType> = {|
 
     // This function returns an object representing the DOM structure to be
     // created when rendering the defined LaTeX function.
-    // TODO: Make return type explicit.
-    htmlBuilder?: (group: ParseNode<NODETYPE>, options: Options) => *,
+    htmlBuilder?: (group: ParseNode<NODETYPE>, options: Options) => HtmlDomNode,
 
     // This function returns an object representing the MathML structure to be
     // created when rendering the defined LaTeX function.
-    // TODO: Make return type explicit.
-    mathmlBuilder?: (group: ParseNode<NODETYPE>, options: Options) => *,
+    mathmlBuilder?: (
+        group: ParseNode<NODETYPE>,
+        options: Options,
+    ) => MathNode | TextNode | domTree.documentFragment,
```

Comment:
Can it return a `documentFragment`?

Reply:
Surprisingly, yes:
https://github.com/Khan/KaTeX/blob/7b22eeb64ad7acf0090a393ba339f044e5886c15/src/functions/op.js#L219

Reply:
If it's used only once, I think we should refactor it or create separate method for MathML as `documentFragment` may have HTML-specific features in the future.

Reply:
Currently, I'm focused primarily on putting a type system in place to capture existing invariants and avoiding modifying any core logic (except where the type system catches actual errors). I've added TODO comments for this refactor separately, and I think it would be easier to do once all of KaTeX is strongly and statically typed.

Reply:
I also think it makes some sense to be able to allow a `documentFragment` return value... they are certainly helpful!  But anyway, we can leave this for future discussion.